hera_cal/vis_clean.py

```diff
@@ -786,6 +787,76 @@ def fft_data(data, delta_bin, wgts=None, axis=-1, window='none', alpha=0.2, edge
     return data, fourier_axes
 
 
+def trim_model(clean_model, clean_resid, dnu, keys=None, noise_thresh=5.0, delay_cut=3000,
+               kernel_size=None, edgecut_low=0, edgecut_hi=0,):
+    """
+    Truncate CLEAN model components below some amplitude threshold.
+
+    Estimate the noise in Fourier space and truncate CLEAN model
+    components below a specified value times the noise.
+
+    Args:
+        clean_model : DataContainer
+            Holds clean_model output of self.vis_clean
+        clean_resid : DataContainer
+            Holds clean_resid output of self.vis_clean
+        dnu : float
+            Frequency channel width [Hz]
+        keys : list of antpairpol tuples
+            List of keys to operate on
+        noise_thresh : float
+            Coefficient times noise to truncate model components below
+        delay_cut : float
+            Minimum |delay| [ns] above which to use in estimating noise
+        kernel_size : int
+            Time median filter kernel_size. None is no median filter.
+        edgecut_low : int
+            Edgecut bins to apply to low edge of frequency band
+        edgecut_hi : int
+            Edgecut bins to apply to high edge of frequency band
+
+    Returns:
+        model : DataContainer
+            Truncated clean_model
+        noise : DataContainer
+            Per integration noise estimate from clean_resid
+    """
+    # get keys
+    if keys is None:
+        keys = [k for k in sorted(set(list(clean_model.keys()) + list(clean_resid.keys()))) if k in clean_model and k in clean_resid]
+
+    # estimate noise in Fourier space
+    model = DataContainer({})
+    noise = DataContainer({})    
+    for k in keys:
+        # get rfft
+        rfft, delays = fft_data(clean_resid[k], dnu, axis=1, window='none', edgecut_low=edgecut_low, edgecut_hi=edgecut_hi, ifft=False, ifftshift=False, fftshift=False)
+        delays *= 1e9
+
+        # get clean_resid NEB
+        neb = noise_eq_bandwidth(~np.isclose(clean_resid[k], 0.0))[:, None]
+
+        # get noise estimate
+        noise[k] = np.median(np.abs((rfft * neb)[:, np.abs(delays) > delay_cut]), axis=1)
+
+        # median filter it
+        if kernel_size is not None:
+            n = noise[k]
+            nlen = len(n)
+            n = np.pad(n, nlen, 'reflect', reflect_type='odd')
+            noise[k] = signal.medfilt(n, kernel_size=kernel_size)[nlen:-nlen]
+
+        # get mfft
+        mfft, _ = fft_data(clean_model[k], dnu, axis=1, window='none', edgecut_low=edgecut_low, edgecut_hi=edgecut_hi, ifft=False, ifftshift=False, fftshift=False)
```

Comment:
Why assume that the window is none? Shouldn't it be the same window you CLEANed with?

Reply:
No, it should be `none` b/c the clean model is ensured to have power only at delay modes of the data (i.e. no leakage) b/c its been CLEANed